### PR TITLE
refactor(StyleSheetList): Use implementation and generated wrapper

### PR DESCRIPTION
**Pull request by ExE-Boss**
Part of https://github.com/jsdom/jsdom/issues/2727.

### CODE REVIEW

File: lib/jsdom/living/cssom/StyleSheetList-impl.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  _remove(sheet) {
+    const { _list } = this;
+
+    const index = _list.indexOf(sheet);
+    if (index >= 0) {
+      _list.splice(index, 1);
+    }
+  }
+
+  _values() {
```
Comment (domenic) [inline lib/jsdom/living/cssom/StyleSheetList-impl.js]: What is this for?
```suggestion
  _values() {
```
```suggestion
  _values() {
```
```suggestion
  _values() {
```

File: lib/jsdom/living/cssom/StyleSheetList.webidl [hunk 2]
```diff
@@ -0,0 +1,6 @@
+[Exposed=Window]
+interface StyleSheetList {
+  [WebIDL2JSValueAsUnsupported=_null]
+  getter CSSStyleSheet? item(unsigned long index);
```
Comment (domenic) [inline lib/jsdom/living/cssom/StyleSheetList.webidl]: Can you put the extended attribute on the same line?

File: lib/jsdom/living/cssom/StyleSheetList.webidl [hunk 3]
```diff
@@ -0,0 +1,6 @@
+[Exposed=Window]
```
Comment (domenic) [inline lib/jsdom/living/cssom/StyleSheetList.webidl]: Can you add a spec link? https://drafts.csswg.org/cssom/#the-stylesheetlist-interface

File: lib/jsdom/living/cssom/StyleSheetList-impl.js (hunk 1 above)
Comment (ExE-Boss) [reply lib/jsdom/living/cssom/StyleSheetList-impl.js]: That’s a leftover from what I think was an attempt at refactoring `forEachMatchingSheetRuleOfElement(…)` in [`style‑rules.js`](https://github.com/jsdom/jsdom/blob/master/jsdom/living/helpers/style-rules.js).

**Pull request merged**